core/server/api/v2/posts.js

```diff
@@ -0,0 +1,179 @@
+const models = require('../../models');
+const common = require('../../lib/common');
+const urlService = require('../../services/url');
+const allowedIncludes = ['created_by', 'updated_by', 'published_by', 'author', 'tags', 'authors', 'authors.roles'];
+const unsafeAttrs = ['author_id', 'status', 'authors'];
+
+module.exports = {
+    docName: 'posts',
+    browse: {
+        options: [
+            'include',
+            'filter',
+            'fields',
+            'formats',
+            'page',
+            'status',
+            'limit',
+            'order',
+            'debug'
+        ],
+        validation: {
+            options: {
+                include: {
+                    values: allowedIncludes
+                },
+                formats: {
+                    values: models.Post.allowedFormats
+                }
+            }
+        },
+        permissions: {
+            unsafeAttrs: unsafeAttrs
+        },
+        query(frame) {
+            return models.Post.findPage(frame.options);
+        }
+    },
+
+    read: {
+        options: [
+            'include',
+            'filter',
+            'fields',
+            'status',
+            'formats',
+            'debug'
+        ],
+        data: [
+            'id',
+            'slug',
+            'status',
+            'uuid'
+        ],
+        validation: {
+            options: {
+                include: {
+                    values: allowedIncludes
+                },
+                formats: {
+                    values: models.Post.allowedFormats
+                }
+            }
+        },
+        permissions: {
+            unsafeAttrs: unsafeAttrs
+        },
+        query(frame) {
+            return models.Post.findOne(frame.data, frame.options)
+                .then((model) => {
+                    if (!model) {
+                        throw new common.errors.NotFoundError({
+                            message: common.i18n.t('errors.api.posts.postNotFound')
+                        });
+                    }
+
+                    return model;
+                });
+        }
+    },
+
+    add: {
+        statusCode: 201,
+        headers: {},
```

Comment:
Why do we need to pass an empty object here? Would it make sense to set a default of `{}` or skip setting headers here - https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/http.js#L40 when they are not defined?

Reply:
Yeah this is not optimal, i did that rly quick, because the headers are dynamic.
i thought about a default schema for the ctrl as well to initialise headers: {}
but i wanted to continue with the next ctrl

Reply:
Makes sense, good to know :)